libbeat/outputs/console/console.go

```diff
@@ -49,21 +50,31 @@ func (c *console) PublishEvent(
 	opts outputs.Options,
 	data outputs.Data,
 ) error {
-	var jsonEvent []byte
+	var serializedEvent []byte
 	var err error
 
-	if c.config.Pretty {
-		jsonEvent, err = json.MarshalIndent(data.Event, "", "  ")
+	if c.config.Format != nil {
+		formattedEvent, err := c.config.Format.Run(data.Event)
+		if err != nil {
+			logp.Err("Fail to format event (%v): %#v", err, data.Event)
+			op.SigCompleted(s)
+			return err
+		}
+		serializedEvent = []byte(formattedEvent)
 	} else {
-		jsonEvent, err = json.Marshal(data.Event)
-	}
-	if err != nil {
-		logp.Err("Fail to convert the event to JSON (%v): %#v", err, data.Event)
-		op.SigCompleted(s)
-		return err
+		if c.config.Pretty {
+			serializedEvent, err = json.MarshalIndent(data.Event, "", "  ")
+		} else {
+			serializedEvent, err = json.Marshal(data.Event)
+		}
+		if err != nil {
+			logp.Err("Fail to convert the event to JSON (%v): %#v", err, data.Event)
+			op.SigCompleted(s)
+			return err
+		}
```

Comment:
maybe error handling can be made a little 'nicer' here. Consider `defer op.SigCompleted(s)` at top of PublishEvent to ensure it's never forgotten.